### PR TITLE
Rcf 127 auto data sync in first login

### PR DESCRIPTION
**Pull request by cherinethailu**
A very short commit message for a wide task. I think unit tests aren't necessary but will dive in if necessary. Manual sync with a button working, auto sync after first log in working. UI wise - The user is notified of the background sync that is in progress.

### CODE REVIEW

File: .github/workflows/build_client.yml [hunk 1]
```diff
@@ -20,10 +20,9 @@ jobs:
       - name: Install npm dependencies
         run: |
           npm install
-      - name: Updating serverBaseURL.
-        run: |
-          find . -type f -name "*build.gradle" -print0 | xargs -0 sed -i "s/api-internal.sandbox.xyz.net/${{ github.event.inputs.defaultServerBaseURL }}/g"
-          find . -type f -name "*gradle.properties" -print0 | xargs -0 sed -i "s/sqp_19c9702eb9f5ddfebbfc36eaa6edf39d22280a49/${{ secrets.SONAR_TOKEN }}/g"
+      - name: Updating libs-snapshot-local to libs-release local for artifactory URL's.
```
Comment (ase-101) [inline .github/workflows/build_client.yml:23]: Why is the sonar token replacement command removed?

File: .github/workflows/push_trigger.yml [hunk 2]
```diff
@@ -19,10 +19,6 @@ jobs:
       - name: Install npm dependencies
         run: |
           npm install
-      - name: Updating serverBaseURL.
-        run: |
-          find . -type f -name "*build.gradle" -print0 | xargs -0 sed -i "s/api-internal.sandbox.xyz.net/${{ github.event.inputs.defaultServerBaseURL }}/g"
```
Comment (ase-101) [inline .github/workflows/push_trigger.yml:24]: Why is this block removed?

File: .github/workflows/push_trigger.yml [hunk 3]
```diff
@@ -64,8 +60,7 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-          SONAR_ORGANIZATION: ${{ secrets.ORG_KEY }}
         run: |
           cd client
           chmod +x gradlew
-          ./gradlew build test testDebugUnitTestCoverage sonarqube --info --warning-mode all
+          ./gradlew build sonarqube --info --warning-mode all
```
Comment (ase-101) [inline .github/workflows/push_trigger.yml:66]: Why is the test removed?

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 4]
```diff
@@ -0,0 +1,676 @@
...
+import io.mosip.registration.keymanager.util.CryptoUtil;
+import io.mosip.registration.packetmanager.util.JsonUtils;
+import io.mosip.registration_client.model.SyncPigeon;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+@Singleton
+public class SyncResponseApi implements SyncPigeon.SyncApi {
```
Comment (ase-101) [inline android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:75]: Can we rename it as MasterDataSyncApi

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 5]
```diff
@@ -0,0 +1,676 @@
...
+                                .setErrorCode("")
+                                .build();
+                        result.success(sync);
+                        return;
+//                        Toast.makeText(context, "User Sync Completed", Toast.LENGTH_LONG).show();
+                    } else
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("UserDetailsSync")
+                                .setSyncProgress(Long.valueOf(3))
+                                .setErrorCode("UserDetailsSync Failed")
```
Comment (ase-101) [inline android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:310]: This should be an errorCode not the errorMessage. Error code could be user_sync_failed and the message for this errorCode to be maintained in I10n files

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 6]
```diff
@@ -0,0 +1,676 @@
...
+                result.success(sync);
+                return;
+            }
+
+            @Override
+            public void onFailure(Call<ResponseWrapper<Map<String, Object>>> call, Throwable t) {
+                sync = new SyncPigeon.Sync.Builder()
+                        .setSyncType("GlobalParamsSync")
+                        .setSyncProgress(Long.valueOf(2))
+                        .setErrorCode("GlobalParamsSync Failed")
```
Comment (ase-101) [inline android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:228]: This should be an errorCode not the errorMessage. Error code could be config_sync_failed and the message for this errorCode to be maintained in I10n files

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 7]
```diff
@@ -0,0 +1,676 @@
...
+                                .setSyncProgress(Long.valueOf(2))
+                                .setErrorCode("")
+                                .build();
+                        result.success(sync);
+                        return;
+                    } else
+                        sync = new SyncPigeon.Sync.Builder()
+                                .setSyncType("GlobalParamsSync")
+                                .setSyncProgress(Long.valueOf(2))
+                                .setErrorCode("GlobalParamsSync Failed")
```
Comment (ase-101) [inline android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:209]: This should be an errorCode not the errorMessage. Error code could be config_sync_failed and the message for this errorCode to be maintained in I10n files

**Review by ase-101 — CHANGES_REQUESTED** [review]

File: ios/Runner/pigeon.m
Comment (ase-101) [inline ios/Runner/pigeon.m:1]: We should not commit this file, this should be generated during build time, lets ignore this file in gitignore

File: ios/Runner/pigeon.h
Comment (ase-101) [inline ios/Runner/pigeon.h:1]: We should not commit this file, this should be generated during build time, lets ignore this file in gitignore

File: .github/workflows/build_client.yml (hunk 1 above)
Comment (cherinethailu) [reply .github/workflows/build_client.yml:23]: I was not aware, it was a conflict emerged upon rebase. I will put it back.

File: .github/workflows/push_trigger.yml (hunk 2 above)
Comment (cherinethailu) [reply .github/workflows/push_trigger.yml:24]: I was not aware, it was a conflict emerged upon rebase. I will put it back.

File: .github/workflows/push_trigger.yml (hunk 3 above)
Comment (cherinethailu) [reply .github/workflows/push_trigger.yml:66]: Do we need both the tests? In that case, will put `test testDebugUnitTestCoverage` back.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java (hunk 4 above)
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:75]: Yes, possible. I will change the class name to `MasterDataSyncApi`.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java (hunk 5 above)
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:310]: Noted. Will change the error message to error code.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java (hunk 6 above)
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:228]: Noted. Will change the error message to error code.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 8]
```diff
@@ -0,0 +1,676 @@
...
+                                .setSyncType("CertificateSync")
+                                .setSyncProgress(Long.valueOf(1))
+                                .setErrorCode("Policy key Sync failed").build();
+                    result.success(sync);
+                    return;
+                } else
+                    sync = new SyncPigeon.Sync.Builder()
+                            .setSyncType("CertificateSync")
+                            .setSyncProgress(Long.valueOf(1))
+                            .setErrorCode("Policy key Sync failed").build();
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:170]: Noted. Will change the error message to error code.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 9]
```diff
@@ -0,0 +1,676 @@
...
+        this.dynamicFieldRepository = dynamicFieldRepository;
+        this.locationRepository = locationRepository;
+        this.blocklistedWordRepository = blocklistedWordRepository;
+        this.syncJobDefRepository = syncJobDefRepository;
+        this.languageRepository = languageRepository;
+        this.jobManagerService = jobManagerService;
+    }
+
+
+    private void syncCertificate(@NonNull SyncPigeon.Result<SyncPigeon.Sync> result) {
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:134]: Will do.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 10]
```diff
@@ -0,0 +1,676 @@
...
+        CenterMachineDto centerMachineDto = getRegistrationCenterMachineDetails();
+        if (centerMachineDto == null)
+            return;
+
+        Call<ResponseWrapper<CertificateResponse>> call = syncRestService.getCertificate(REG_APP_ID,
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:139]: Will do.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 11]
```diff
@@ -0,0 +1,676 @@
...
+                return;
+            }
+
+            @Override
+            public void onFailure(Call<ResponseWrapper<CertificateResponse>> call, Throwable t) {
+                sync = new SyncPigeon.Sync.Builder()
+                        .setSyncType("CertificateSync")
+                        .setSyncProgress(Long.valueOf(1))
+                        .setErrorCode("Policy key Sync failed").build();
+                result.success(sync);
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:181]: Noted. Will change the error message to error code.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 12]
```diff
@@ -0,0 +1,676 @@
...
+                                .setSyncProgress(Long.valueOf(3))
+                                .setErrorCode("UserDetailsSync Failed")
+                                .build();
+                    result.success(sync);
+                    return;
+                } else
+                    sync = new SyncPigeon.Sync.Builder()
+                            .setSyncType("UserDetailsSync")
+                            .setSyncProgress(Long.valueOf(3))
+                            .setErrorCode("UserDetailsSync Failed")
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:318]: Noted. Will change the error message to error code.

File: android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java [hunk 13]
```diff
@@ -0,0 +1,676 @@
...
+                result.success(sync);
+                return;
+            }
+
+            @Override
+            public void onFailure(Call<ResponseWrapper<UserDetailResponse>> call, Throwable t) {
+                sync = new SyncPigeon.Sync.Builder()
+                        .setSyncType("UserDetailsSync")
+                        .setSyncProgress(Long.valueOf(3))
+                        .setErrorCode("UserDetailsSync Failed")
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/app/src/main/java/io/mosip/registration_client/api_services/SyncResponseApi.java:329]: Noted. Will change the error message to error code.

File: ios/Runner/pigeon.m
Comment (cherinethailu) [reply ios/Runner/pigeon.m:1]: Will do.

File: android/keymanager/src/main/assets/config.properties [hunk 14]
```diff
@@ -17,9 +17,9 @@ mosip.kernel.crypto.asymmetric-algorithm-message-digest-function=SHA-256
 #Crypto asymmetric algorithm OAEP mask generation function
 mosip.kernel.crypto.asymmetric-algorithm-mask-generation-function=MGF1
 #Crypto symmetric IV length
-mosip.kernel.crypto.symmetric-algorithm-iv-length=12
+mosip.kernel.crypto.symmetric-algorithm-iv-length=16
```
Comment (jainhitesh9998) [inline android/keymanager/src/main/assets/config.properties:20]: any specific reason for property change here

Comment (cherinethailu) [reply android/keymanager/src/main/assets/config.properties:20]: That was a quick fix I got from Piyush for login. Not sure about the details. No specific reason.

File: android/keymanager/src/main/assets/config.properties [hunk 15]
```diff
@@ -17,9 +17,9 @@ mosip.kernel.crypto.asymmetric-algorithm-message-digest-function=SHA-256
...
 #Crypto symmetric AAD length
-mosip.kernel.crypto.symmetric-algorithm-aad-length=32
+mosip.kernel.crypto.symmetric-algorithm-aad-length=12
```
[the remark this answers is not in the document]
Comment (cherinethailu) [reply android/keymanager/src/main/assets/config.properties:22]: That was a quick fix I got from Piyush for login. Not sure about the details. No specific reason.

File: .github/workflows/build_client.yml (hunk 1 above)
Comment (jainhitesh9998) [reply .github/workflows/build_client.yml:23]: can you revert this file as well

File: .github/workflows/push_trigger.yml (hunk 2 above)
Comment (jainhitesh9998) [reply .github/workflows/push_trigger.yml:24]: can you revert this file as well

File: .github/workflows/push_trigger.yml (hunk 3 above)
Comment (jainhitesh9998) [reply .github/workflows/push_trigger.yml:70]: can you revert this as well

Comment (cherinethailu) [reply .github/workflows/push_trigger.yml:70]: image
This is what I see from my local latest update

File: .github/workflows/push_trigger.yml (hunk 2 above)
Comment (cherinethailu) [reply .github/workflows/push_trigger.yml:24]: image

File: .github/workflows/build_client.yml (hunk 1 above)
Comment (cherinethailu) [reply .github/workflows/build_client.yml:23]: image

**Review by jainhitesh9998 — APPROVED** [review]

**Pull request merged**